test/parallel/test-timers-unref-reuse-no-exposed-list.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+const timer1 = setTimeout(() => {}, 1).unref();
+assert.strictEqual(timer1._handle._list, undefined,
+                   'timer1._handle._list should be undefined');
+
+// Check that everything works even if the handle was not re-used.
+setTimeout(() => {}, 1);
+const timer2 = setTimeout(() => {}, 1).unref();
+// Note: It is unlikely that this assertion will be hit in a failure.
```

Comment:
The test code itself is great, and the comment above:

> // Check that everything works even if the handle was not re-used.

is both necessary and sufficient to explain why this test code is present.

On the other hand,  I find the comment:

> // Note: It is unlikely that this assertion will be hit in a failure.
> // Instead, timers.js will likely throw a TypeError internally.

confusing and detrimental to the understanding of that test because it does not reflect the current state of the implementation. Instead it describes what could happen if the code was changed in a specific way, that is if a conditional was removed.

However, I don't think we want to start documenting every possible way this test could fail depending on an infinite list of possible changes, or by definition we will end up with an infinite list of comments.